### DISCUSSION TITLE
How to shown values of arrayfire array during debugging?

### DISCUSSION THREAD

**Discussion by 3togo** [arrayfire/arrayfire-rust#345] [Q&A] [answered]
Most of the rust compatible variables could reveal their values during debugging. However arrayfire array will only show something like
"handle: core::ffi::c_void::__variant1" 
without any details of their stored value.

I have tried the cpu backend. The situation is the same. I wonder whether there is any mean to show the value of arrayfire array during code debugging?

**Comment by 9prady9** [comment, marked as answer]
You should be able to view debugging information using `gdb`, or `lldb` I think (I haven't tried lldb myself, but should work). Which tool are you using ?

Also, please do note that, even with gdb you still won't be able to view array contents itself directly in a debugger since gdb/lldb or any debugger for that matter cannot provide you a view of the GPU memory. You can use the [list of functions](https://arrayfire.org/arrayfire-rust/arrayfire/index.html?search=print) for viewing the array contents. Metadata, such as dimensions, type of data etc., of any given error would be visible in debugger though.
